Content.Server/SubFloor/SubFloorHideSystem.cs

```diff
@@ -1,16 +1,46 @@
 using Content.Shared.Construction.Components;
+using Content.Shared.Eye;
 using Content.Shared.SubFloor;
+using Robust.Server.Console;
+using Robust.Shared.Console;
 using Robust.Shared.Map.Components;
 
 namespace Content.Server.SubFloor;
 
 public sealed class SubFloorHideSystem : SharedSubFloorHideSystem
 {
+    [Dependency] private readonly IConGroupController _console = default!;
+    [Dependency] private readonly SharedEyeSystem _eye = default!;
+
     public override void Initialize()
     {
         base.Initialize();
         SubscribeLocalEvent<SubFloorHideComponent, AnchorAttemptEvent>(OnAnchorAttempt);
         SubscribeLocalEvent<SubFloorHideComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
+        SubscribeNetworkEvent<ShowSubfloorRequestEvent>(OnShowSubfloor);
+    }
+
+    private void OnShowSubfloor(ShowSubfloorRequestEvent ev, EntitySessionEventArgs args)
+    {
+        // TODO: Commands are a bit of an eh? for client-only but checking shared perms
+        var ent = args.SenderSession.AttachedEntity;
+
+        if (!TryComp(ent, out EyeComponent? eyeComp))
+            return;
+
+        if (ev.Value)
+        {
+            _eye.SetVisibilityMask(ent.Value, eyeComp.VisibilityMask | (int) VisibilityFlags.Subfloor, eyeComp);
```

Comment:
```suggestion
            _eye.SetVisibilityMask(ent.Value, eyeComp.VisibilityMask | (int)VisibilityFlags.Subfloor, eyeComp);
```
Is your editorconfig correct?
You are always doing the casts wrong. This should be a warning in your IDE.

Reply:
It's not an editor warning, I don't recall ever touching it in the past 5 years. I don't really see an issue with it and this is what it defaults to (the green is rider saying to use nulls not defaults to be more explicit).

![image](https://github.com/user-attachments/assets/d5cabc58-27de-462a-95e4-292eb1aa34d2)